tests/testSerializationSlam.cpp

```diff
@@ -52,6 +52,8 @@
 #include <gtsam/base/serializationTestHelpers.h>
 #include <gtsam/base/std_optional_serialization.h>
 
+#ifdef GTSAM_USE_BOOST_FEATURES
```

Comment:
I think this can be removed since we exclude this test file using CMake if boost serialization is disabled. 

Ref: tests/CMakeLists.txt

Does this test file still get compiled even with the boost serialization flag set to OFF?

Reply:
Yes, with both USE_BOOST_FEATURES and ENABLE_BOOST_SERIALIZATION set to OFF, it still attempts to compile this test. 

I think I see the problem. the CMakeLists.txt tries to exclude "testSerializationSLAM.cpp" (with SLAM in all caps) instead of "testSerializationSlam.cpp"

Reply:
That seems plausible. I can't reproduce the issue on my end but that may be because of differences in CMake versions or the platform (I'm using MacOS).

Can you please update the CMakeLists.txt file? Then we can remove this guard.

Reply:
OK, I have update the CMakeLists.txt file and removed the guard, and now it works on my end.